src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseDSA.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotUseDSA : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA5384";
+        private static readonly LocalizableString s_Title = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.DoNotUseDSA),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.DoNotUseDSAMessage),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.DoNotUseDSADescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTagsExtensions.DataflowAndTelemetry);
+
+        private static readonly ImmutableHashSet<string> s_DSAAlgorithmNames =
+            ImmutableHashSet.Create(
+                "DSA",
+                "System.Security.Cryptography.DSA");
```

Comment:
Are these case sensitive?  If not, may want to initialize with StringComparer.OrdinalIgnoreCase.

Reply:
I mean:
```csharp
ImmutableHashSet.Create(
    StringComparer.OrdinalIgnoreCase,
    "DSA",
    "System.Security.Cryptography.DSA");
```

Then you can just use `Contains()` below, instead of having to enumerate.